fgpyo/sam/__init__.py

```diff
@@ -1266,8 +1190,220 @@ class SamOrder(enum.Enum):
 
 
 @dataclass(frozen=True)
-class SecondaryAlignment:
-    """A secondary alignment as encoded in one part of the `XA` or `XB` tag value on a SAM record.
+class AuxAlignment(ABC, DataclassInstance):
+    """An alignment parsed from the data stored in the SAM tags of a SAM record.
```

Comment:
This is typically where it's parsed from, but really, it's just some minimum information about an alternative alignment.  Would you be ope to changing the description?

Reply:
I see what you mean, but the implementation is heavily hardcoded around the idea of building auxiliary alignments from SAM tags. For example, the fields are a rough subset of what you have for a pysam alignment, but only those that exist in SAM tags. I'm leaving the description as-is for now and we can revisit once I finish a refactor to see if rephrasing it makes more sense to me.